v2/internal/testutils/testutils.go

```diff
@@ -122,3 +131,276 @@ func LocalMirrorFromFiles(source string, destination string) error {
+		return false, err
+	}
+	if desc != nil {
+		return true, nil
+	}
+	return false, nil
+}
+
+type releaseContents struct {
+	Ref1 string
+	Ref2 string
+	Ref3 string
+}
+
+func GenerateReleaseAndComponents(toRegistry, tempFolder string, templatePath string) (string, []string, error) {
+	contents := releaseContents{}
+	relatedImages := []string{}
+	// generating a fake release with 3 components
+	component1 := toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev:component1"
+	digest1, err := GenerateFakeImage("component1", component1, tempFolder)
+	if err != nil {
+		return "", relatedImages, err
+	}
+	contents.Ref1 = toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev@" + digest1
+	relatedImages = append(relatedImages, contents.Ref1)
+
+	component2 := toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev:component2"
+	digest2, err := GenerateFakeImage("component2", component2, tempFolder)
+	if err != nil {
+		return "", relatedImages, err
+	}
+	contents.Ref2 = toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev@" + digest2
+	relatedImages = append(relatedImages, contents.Ref2)
+
+	component3 := toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev:component3"
+	digest3, err := GenerateFakeImage("component3", component3, tempFolder)
+	if err != nil {
+		return "", relatedImages, err
+	}
+	contents.Ref3 = toRegistry + "/openshift-release-dev/ocp-v4.0-art-dev@" + digest3
+	relatedImages = append(relatedImages, contents.Ref3)
+
+	digest, err := GenerateFakeRelease(contents, toRegistry+"/openshift-release-dev/ocp-release:4.15.0-x86_64", tempFolder, templatePath)
+	if err != nil {
+		return "", relatedImages, err
+	}
+	relatedImages = append(relatedImages, toRegistry+"/openshift-release-dev/ocp-release@"+digest)
+	return digest, relatedImages, nil
+}
+
+func GenerateFakeRelease(imageRefs releaseContents, releaseImgRef, tempFolder string, templateFile string) (string, error) {
+	err := ImgRefsFromTemplate(tempFolder+"/image-references", templateFile, imageRefs)
+	if err != nil {
+		return "", err
+	}
+	imgSpec, err := image.ParseRef(releaseImgRef)
+	if err != nil {
+		return "", err
+	}
+	sanitizedTagOrDigest := imgSpec.Tag
+	if imgSpec.Tag == "" {
+		sanitizedTagOrDigest = imgSpec.Digest
+	} else {
+		sanitizedTagOrDigest = strings.ReplaceAll(sanitizedTagOrDigest, ".", "-")
+	}
+
+	indexFolder := filepath.Join(tempFolder, imgSpec.PathComponent, sanitizedTagOrDigest)
+	err = os.MkdirAll(indexFolder, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	imageRefsBytes, err := os.ReadFile(tempFolder + "/image-references")
+	if err != nil {
+		return "", err
+	}
+	releaseMetaBytes, err := os.ReadFile("../../internal/e2e/templates/release_templates/release-metadata")
+	if err != nil {
+		return "", err
+	}
+	c := map[string][]byte{
+		"/release-manifests/image-references": imageRefsBytes,
+		"/release-manifests/release-metadata": releaseMetaBytes,
+	}
+	tag, err := name.NewTag(releaseImgRef)
+	if err != nil {
+		return "", err
+	}
+	i, _ := crane.Image(c)
```

Comment:
@sherine-k - looks like some duplication here - maybe create a simple common utility

Reply:
That's very good point thanks!